///////////////////////////////////////////////////////////////////////////////
/// Headers //////////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////

#include "ServerPanelMainWindow.h"
#include "ui_ServerPanelMainWindow.h"

///////////////////////////////////////////////////////////////////////////////
/// Globals //////////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////

ServerPanelMainWindow* ServerPanelMainWindow::mInstance = NULL;

///////////////////////////////////////////////////////////////////////////////
/// Singleton ////////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////

ServerPanelMainWindow* ServerPanelMainWindow::Instance() {
    // Check for an existing instance
    if (!mInstance) {
        // Create a new instance
        mInstance = new ServerPanelMainWindow();
    }
    // Return the instance
    return mInstance;
}

///////////////////////////////////////////////////////////////////////////////
/// Constructor //////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////

ServerPanelMainWindow::ServerPanelMainWindow(QWidget *cParent) : QWidget(cParent), mUserInterface(new Ui::ServerPanelMainWindow) {
    // Setup the user interface object
    this->mUserInterface->setupUi(this);
    // Setup a Sign-In button click handler
    connect(this->mUserInterface->btnSignIn, SIGNAL(clicked()), this, SLOT(TrySignIn()));
}

///////////////////////////////////////////////////////////////////////////////
/// Destructor ///////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////

ServerPanelMainWindow::~ServerPanelMainWindow() {
    // Delete the user interface object
    delete this->mUserInterface;
}

///////////////////////////////////////////////////////////////////////////////
/// Event Handlers ///////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////

void ServerPanelMainWindow::TrySignIn () {
    // Create an error place holder
    QString sErrorText;
    // Check for valid data
    if (this->mUserInterface->txtUsername->text().isEmpty()) {
        // Add to the error string
        sErrorText += "<font color=\"red\">Missing username.</font>\n";
    }
    if (this->mUserInterface->txtPassword->text().isEmpty()) {
        // Add to the error string
        sErrorText += "<font color=\"red\">Missing password.</font>\n";
    }
    // Do we have errors
    if (!sErrorText.isEmpty()) {
        // Show the message box
        QMessageBox::information(this, "Error!", sErrorText);
    }
}
